src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufMetricExporter.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Runtime.InteropServices;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal class OtlpProtobufMetricExporter : IDisposable
+{
+    private const int BufferSize = 65360; // the maximum ETW payload (inclusive)
+
+    private readonly byte[] buffer = new byte[BufferSize];
+
+    private readonly OtlpProtobufSerializer otlpProtobufSerializer;
+
+    public OtlpProtobufMetricExporter()
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // Temporary until we add support for user_events.
+            throw new NotSupportedException("Unix domain socket should not be used on Windows.");
+        }
+
+        this.otlpProtobufSerializer = new OtlpProtobufSerializer(MetricEtwDataTransport.Instance);
+    }
+
+    public ExportResult Export(in Batch<Metric> batch, Resource resource = null)
+    {
+        var result = ExportResult.Success;
+
+        int currentPosition = 0;
+
+        try
+        {
+            this.otlpProtobufSerializer.SerializeAndSendMetrics(this.buffer, ref currentPosition, resource, batch);
```

Comment:
Could you split the responsibilities here? `OtlpProtobufSerializer` class and its methods should only be responsible for serializing the data and getting the `byte[]` buffer ready. You could then call `MetricEtwDataTransport.Send` separately after you have serialized the data.

Reply:
@utpilla Just adding the notes based on offline discussion. With the current state it is possible to do so as we would be serializing 1 metricPoint at a time. However, with the follow up I am planning to do (sending as much data as possible in one etw event), it would get coupled again. There isn't a way to completely decouple the serializer with sender.

There was one concern I had with regards to testing/benchmarking, but that can be managed by setting the transport to no-op. LMK what you think.